pkg/bot/mattermost.go

```diff
@@ -315,100 +299,105 @@ func (b *Mattermost) listen(ctx context.Context) {
-
-		if isDefaultChannel {
-			return createPostWrappedErr
+	// TODO(https://github.com/kubeshop/botkube/issues/596): Support source bindings - filter events here or at source level and pass it every time via event property?
+	var channelsToNotify []string
+	for _, channelCfg := range b.getChannels() {
+		if !channelCfg.notify {
+			b.log.Info("Skipping notification for channel %q as notifications are disabled.", channelCfg.Identifier())
+			continue
 		}
 
-		// fallback to default channel
+		channelsToNotify = append(channelsToNotify, channelCfg.Identifier())
+	}
+	return channelsToNotify
+}
 
-		// send error message to default channel
-		msg := fmt.Sprintf("Unable to send message to channel %q: `%s`\n```add Botkube app to the channel %q\nMissed events follows below:```", targetChannel, resp.Error, targetChannel)
-		sendMessageErr := b.SendMessage(ctx, msg)
-		if sendMessageErr != nil {
-			return multierror.Append(createPostWrappedErr, sendMessageErr)
+// SendMessage sends message to Mattermost channel
+func (b *Mattermost) SendMessage(_ context.Context, msg string) error {
+	var errs error
+	for _, channel := range b.getChannels() {
+		channelID := channel.ID
+		b.log.Debugf(">> Sending message to channel %q: %+v", channelID, msg)
+		post := &model.Post{
+			ChannelId: channelID,
+			Message:   msg,
 		}
-
-		// sending missed event to default channel
-		// reset channel, so it will be defaulted
-		event.Channel = ""
-		sendEventErr := b.SendEvent(ctx, event)
-		if sendEventErr != nil {
-			return multierror.Append(createPostWrappedErr, sendEventErr)
+		if _, resp := b.apiClient.CreatePost(post); resp.Error != nil {
+			errs = multierr.Append(errs, fmt.Errorf("while creating a post: %w", resp.Error))
 		}
 
-		return createPostWrappedErr
+		b.log.Debugf("Message successfully sent to channel %q", channelID)
 	}
+	return errs
+}
 
-	b.log.Debugf("Event successfully sent to channel %q", post.ChannelId)
-	return nil
+func (b *Mattermost) getChannels() map[string]channelConfigByID {
+	b.channelsMutex.RLock()
+	defer b.channelsMutex.RUnlock()
+	return b.channels
 }
 
-// SendMessage sends message to Mattermost channel
-func (b *Mattermost) SendMessage(_ context.Context, msg string) error {
-	post := &model.Post{
-		ChannelId: b.ChannelID,
-		Message:   msg,
-	}
-	if _, resp := b.APIClient.CreatePost(post); resp.Error != nil {
-		return fmt.Errorf("while creating a post: %w", resp.Error)
+func (b *Mattermost) setChannels(channels map[string]channelConfigByID) {
+	b.channelsMutex.Lock()
+	defer b.channelsMutex.Unlock()
+	b.channels = channels
+}
+
+func mattermostChannelsCfgFrom(client *model.Client4, teamID string, channelsCfg config.IdentifiableMap[config.ChannelBindingsByName]) (map[string]channelConfigByID, error) {
+	res := make(config.IdentifiableMap[channelConfigByID])
+	for _, channCfg := range channelsCfg {
+		fetchedChannel, resp := client.GetChannelByName(channCfg.Name, teamID, "")
```

Comment:
```suggestion
		fetchedChannel, resp := client.GetChannelByName(channCfg.Identifier(), teamID, "")
```
let's be consistent, use `Identifier()` in all places or explicit field value. If we have a bug, it will be easy to trace and fix in all places.